cdap-common/src/main/java/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java

```diff
@@ -50,21 +54,46 @@
  * </p>
  */
 // TODO: This is a copy from Twill, and can be removed when TWILL-119 is fixed. CDAP-4923 tracks Twill-0.8.0 upgrade.
+// After copying, we have removed the addRMToken() method. That functionality is in YarnTokenUtils.
```

Comment:
Since this is a class copied from Twill, do you have a corresponding twill ticket? You should include it in the TODO as well.